src/llm/default.py

```diff
@@ -0,0 +1,99 @@
+import logging
+
+from llm.abstract_llm import AbstractLLM
+from llm.memory import generate_memory_instance
+from llm.embeddings import get_most_relevant_contents_from_message
+
+from langchain.chains.llm import LLMChain
+from langchain.memory.chat_memory import BaseChatMemory
+
+from langchain.chat_models import ChatOpenAI
+from langchain.memory import ConversationBufferWindowMemory
+from langchain.prompts import (ChatPromptTemplate, HumanMessagePromptTemplate,
+                               MessagesPlaceholder,
+                               SystemMessagePromptTemplate)
+
+from settings import OPENAI_API_KEY
+
+
+class DialogLLM(AbstractLLM):
+    @property
+    def memory(self) -> BaseChatMemory:
+        if self.session_id:
+            return generate_memory_instance(
+                session_id=self.session_id,
+                parent_session_id=self.config.get("parent_session_id")
+            )
+        return None
+
+    def preprocess(self, input: str) -> str:
+        return input
```

Comment:
You can ignore the `preprocess` and `postprocess` methods because they use the same implementation on base class.

Reply:
i can remove the pre-process, but on post-process i'll do the asyncio create_task on the categorization